vllm/prompt_adapter/layers.py

```diff
@@ -0,0 +1,56 @@
+from dataclasses import dataclass
+from typing import Dict, List, Optional
+
+import numpy
+import torch
+from torch import nn
+
+from vllm.adapter_commons.layers import AdapterMapping
+from vllm.model_executor.layers.vocab_parallel_embedding import (
+    VocabParallelEmbedding)
+
+
+@dataclass
+class PromptAdapterMapping(AdapterMapping):
+    pass
+
+
+class VocabParallelEmbeddingWithPromptAdapter(nn.Module):
```

Comment:
Looking at this code it makes me thing it will clash with LoRA adapters (either the LoRA or this will not be ran at the same time if both are specified).

Reply:
It doesn't seem to do this, since it extends the lora class which happens before this.
I have added a new test which tests having both lora and pa specified in the same request.